docs/cars.py

```diff
@@ -0,0 +1,114 @@
+    # TODO: add YouTube videos
+    # Some minimum steering speeds are not yet in CarParams
+    min_steer_speed = CP.minSteerSpeed
+    if car_info.min_steer_speed is not None:
+      min_steer_speed = car_info.min_steer_speed
+      assert CP.minSteerSpeed == 0, "Minimum steer speed set in both CarInfo and CarParams for {}".format(CP.carFingerprint)
+
+    min_enable_speed = CP.minEnableSpeed
+    if car_info.min_enable_speed is not None:
+      min_enable_speed = car_info.min_enable_speed
+
+    stars = [CP.openpilotLongitudinalControl and not CP.radarOffCan, min_enable_speed <= 0., min_steer_speed <= 0.,
+             CP.carName in MAKES_GOOD_STEERING_TORQUE, CP.carFingerprint not in non_tested_cars]
+    row = [self.make, self.model, car_info.package, *map(lambda star: "full" if star else "empty", stars)]
+
+    # Check for car footnotes and star demotions
+    star_count = 0
+    for row_idx, column in enumerate(Column):
+      footnote = get_footnote(car_info, column)
+      if column in StarColumns:
+        # Demote if footnote specifies a star
+        if footnote is not None and footnote.value.star is not None:
+          row[row_idx] = footnote.value.star
+        star_count += row[row_idx] == "full"
+        row[row_idx] = get_star_icon(row[row_idx])
+
+      if footnote is not None:
+        row[row_idx] += "<sup>{}</sup>".format(ALL_FOOTNOTES[footnote])
+
+    return row, star_count
+
+
+def get_tiered_cars():
+  # Keep track of cars while sorting by make, model name, and year
+  tiered_cars = {tier: SortedList(key=lambda car: car.make + car.model) for tier in Tier}
+
+  for _, models in get_interface_attr("CAR_INFO").items():
+    for model, car_info in models.items():
+      # Hyundai exception: all have openpilot longitudinal
+      fingerprint = defaultdict(dict)
+      fingerprint[1] = {HKG_RADAR_START_ADDR: 8}
+      CP = interfaces[model][0].get_params(model, fingerprint=fingerprint)
+      # Skip community supported
+      if CP.dashcamOnly:
+        continue
+
+      # Some candidates have multiple variants
+      if not isinstance(car_info, list):
+        car_info = [car_info]
+
+      for _car_info in car_info:
+        car = Car(_car_info, CP)
+        tiered_cars[car.tier].add(car)
+
+  # Return tier name and car rows for each tier
+  for tier, cars in tiered_cars.items():
+    yield [tier.name.title(), map(lambda car: car.row, cars)]
+
+
+def generate_cars_md(tiered_cars):
+  template_fn = os.path.join(BASEDIR, "docs", "CARS_template.md")
+  with open(template_fn, "r") as f:
+    template = jinja2.Template(f.read(), trim_blocks=True)
+
+  footnotes = map(lambda fn: fn.value.text, ALL_FOOTNOTES)
+  return template.render(tiers=tiered_cars, columns=[column.value for column in Column], footnotes=footnotes)
+
+
+if __name__ == "__main__":
+  # Auto generates supported cars documentation
+  # Cars that can disable radar have openpilot longitudinal
+  Params().put_bool("DisableRadar", True)
```

Comment:
this is pretty hacky, I think we want to move all the params usages out of selfdrive/car (in another PR)